streams/test-utils/src/main/java/org/apache/kafka/streams/TopologyTestDriver.java

```diff
@@ -1058,6 +1072,11 @@ public void close() {
             }
         }
         completeAllProcessableWork();
+        if (task.hasRecordsQueued()) {
+            log.warn("Due to the {} configuration, there were some records that can't be processed even" +
```

Comment:
`can't` -> `could not` ?

I would remove `even though TopologyTestDriver is shutting down` (read cleaner this way IMHO.)